src/embed/connection.rs

```diff
@@ -309,12 +315,23 @@ async fn do_open_transaction<'a, 'b>(
 ) -> Result<OpenTransactionResponse, OpenTransactionError> {
     use OpenTransactionError::*;
 
+    // We create a Subscription object to keep track of what keys and scans were
+    // read in this subscribe-query transaction.
+    let subscription = if req.is_subscription {
+        let inner = SubscriptionInner::new();
```

Comment:
@phritz Turns out that I didn't need the global map. JS needs to maintain the subscription. The transaction owns the Subscription instance and it goes away when the transaction goes away.

Reply:
After @aboodman hint I moved the whole subscription tracking to JS